frontend/model/contracts/chatroom.js

```diff
@@ -674,13 +673,13 @@ sbp('chelonia/defineContract', {
         console.warn(`rotateKeys: ${e.name} thrown during queueEvent to ${contractID}:`, e)
       })
     },
-    'gi.contracts/chatroom/removeForeignKeys': (contractID, memberID, state) => {
-      const keyIds = findForeignKeysByContractID(state, memberID)
+    'gi.contracts/chatroom/removeForeignKeys': async (contractID, memberID, state) => {
+      const keyIds = await sbp('chelonia/contract/foreignKeysByContractID', state, memberID)
 
       if (!keyIds?.length) return
 
-      const CSKid = findKeyIdByName(state, 'csk')
-      const CEKid = findKeyIdByName(state, 'cek')
+      const CSKid = await sbp('chelonia/contract/currentKeyIdByName', state, 'csk', true)
+      const CEKid = await sbp('chelonia/contract/currentKeyIdByName', state, 'cek')
```

Comment:
Why were these changes made?

Reply:
Better separation, and it reduces contract size by about ~150 KiB (this likely due to poor tree shaking)